projects/netgrif-components-core/src/lib/impersonation/services/impersonation.service.ts

```diff
@@ -0,0 +1,96 @@
+import {Injectable, OnDestroy} from '@angular/core';
+import {Router} from '@angular/router';
+import {TranslateService} from '@ngx-translate/core';
+import {AbstractResourceService, ResourceProvider, UserResource} from '../../resources/public-api';
+import {ConfigurationService} from '../../configuration/configuration.service';
+import {UserService} from '../../user/services/user.service';
+import {SnackBarService} from '../../snack-bar/services/snack-bar.service';
+import {FilterRepository} from '../../filter/filter.repository';
+import {LoggerService} from '../../logger/services/logger.service';
+import {Observable, Subject, Subscription} from 'rxjs';
+import {User} from '../../user/models/user';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ImpersonationService extends AbstractResourceService implements OnDestroy {
+
+    private _impersonating$ = new Subject<boolean>();
+
+    private _lastUser: User;
+    private _sub: Subscription;
+
+    public constructor(protected provider: ResourceProvider,
+                       protected _router: Router,
+                       protected _configService: ConfigurationService,
+                       protected _userService: UserService,
+                       protected _snackbar: SnackBarService,
+                       protected _filter: FilterRepository,
+                       protected _log: LoggerService,
+                       private _translate: TranslateService) {
```

Comment:
can we change this property to protected in case of extending this service in some project ?

Reply:
done